libopenage/gamestate/game_main.cpp

```diff
@@ -37,17 +38,14 @@ GameMain::GameMain(const Generator &generator)
 }
 
 GameMain::~GameMain() {
-	for (auto &p : this->players) {
-		delete p;
-	}
 }
 
 unsigned int GameMain::player_count() const {
 	return this->players.size();
 }
 
 Player *GameMain::get_player(unsigned int player_id) {
-	return this->players.at(player_id);
+	return this->players.at(player_id).get();
```

Comment:
I think you are not in my branch, you must be on the master. 
The get_player was not used and the vector was public (wrong design, you shouldn't be able to change the player list), so I made private and the used the get_player.
Now its used here:
````
libopenage/gamestate/generator.cpp
libopenage/gamestate/game_main.cpp
libopenage/gamestate/game_save.cpp
libopenage/gamestate/game_main.h
libopenage/game_control.cpp
````
And then all of these pass the Player* here:
```
libopenage/gamestate/score.cpp
libopenage/gamestate/score.h
libopenage/gamestate/team.cpp
libopenage/gamestate/team.h
libopenage/game_control.cpp
libopenage/game_control.h
```